src/SoftRobots/component/behavior/SoftRobotsConstraint.h

```diff
@@ -115,37 +111,39 @@ class SOFA_SOFTROBOTS_API SoftRobotsConstraint : public SoftRobotsBaseConstraint
-    virtual void buildConstraintMatrix(const ConstraintParams* cParams,
+    virtual void buildConstraintMatrix(const sofa::core::ConstraintParams* cParams,
                                        DataMatrixDeriv &c,
                                        unsigned int &cIndex,
                                        const DataVecCoord &x) = 0;
 
 
-    void storeLambda(const ConstraintParams* cParams, MultiVecDerivId res, const linearalgebra::BaseVector* lambda) override;
+    void storeLambda(const sofa::core::ConstraintParams* cParams, sofa::core::MultiVecDerivId res, const sofa::linearalgebra::BaseVector* lambda) override;
 
 
 protected:
-    Data<Real> d_endTime;  ///< Time when the constraint becomes inactive (-1 for infinitely active)
+    sofa::Data<Real> d_endTime;  ///< Time when the constraint becomes inactive (-1 for infinitely active)
 
-    SoftRobotsConstraint(MechanicalState<DataTypes> *mm = NULL);
+    SoftRobotsConstraint(sofa::core::behavior::MechanicalState<DataTypes> *mm = nullptr);
     ~SoftRobotsConstraint() override;
 
-    MechanicalState<DataTypes> *m_state; ///< Associated mechanical state
+    sofa::core::behavior::MechanicalState<DataTypes> *m_state { nullptr }; ///< Associated mechanical state
 
 private:
-    void storeLambda(const ConstraintParams* cParams, Data<VecDeriv>& resId, const Data<MatrixDeriv>& jacobian, const sofa::linearalgebra::BaseVector* lambda);
+    void storeLambda(const sofa::core::ConstraintParams* cParams, sofa::Data<VecDeriv>& resId, const sofa::Data<MatrixDeriv>& jacobian, const sofa::linearalgebra::BaseVector* lambda);
 };
 
 
-// Force template specialization for the most common sofa float related type.
-// This goes with the extern template declaration in the .h. Declaring extern template
-// avoid the code generation of the template for each compilation unit.
-// see: http://www.stroustrup.com/C++11FAQ.html#extern-templates
+#if !defined(SOFTROBOTS_SOFTROBOTSCONSTRAINT_CPP)
+extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<sofa::defaulttype::Vec3Types>;
+extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<sofa::defaulttype::Vec2Types>;
+extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<sofa::defaulttype::Vec1Types>;
+extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<sofa::defaulttype::Rigid3Types>;
+#endif
+} // namespace SoftRobots
 
-extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<defaulttype::Vec3Types>;
-extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<defaulttype::Vec2Types>;
-extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<defaulttype::Vec1Types>;
-extern template class SOFA_SOFTROBOTS_API SoftRobotsConstraint<defaulttype::Rigid3Types>;
-
-} // namespace sofa::core::behavior
+namespace sofa::core::behavior
+{
+    template <class DataTypes>
+    using SoftRobotsConstraint = SoftRobots::SoftRobotsConstraint<DataTypes>;
+}
```

Comment:
Could you explain me why we should do this? 
Otherwise, lowercase maybe for the namespace? to follow sofa's naming code style.

Reply:
Earlier in the file, I changed the namespace of the class `SoftRobotsConstraint` from `sofa::core::behavior` to `SoftRobots`. It has been considered that a good practice for a plugin is to name the namespaces based on the plugin name. You're probably right that it could be in lowercase.
The line you are mentioning is to keep compatibility with the previous namespace. The class `SoftRobotsConstraint` is defined in `SoftRobots`, but it is still usable in `sofa::core::behavior`. This is to not break dependent plugins such as Inverse or Cosserat.

Reply:
Okay so this is just temporary. 
I think the namespace is a good idea indeed, I was just confused by this step backward with no comment or deprecated tag or tool.

Reply:
Also I would keep something like:
- softrobots::behavior
- softrobots::constraint
- softrobots::forcefield
- etc.

Reply:
I added the deprecation mechanism. I did not do it initially because I doubted that it was possible to deprecated a templated alias. It seems that it is possible but not all compilers support it. Let's see if the CI agrees.

Reply:
> Also I would keep something like:
> 
> * softrobots::behavior
> * softrobots::constraint
> * softrobots::forcefield
> * etc.

why not, but this PR is not affected, right?

Reply:
For this PR softrobots::behavior